analysis/analysis.py

```diff
@@ -34,6 +34,96 @@ def load_data_csv(csv_path: Path):
     return out_data
 
 
+def fetch_responses(dataset):
+    """
+    Extracts text responses to "In three words or fewer, what is this text about?" from given csv
+    file, depending on context given (story vs. ad). Returns lists of words used in question
+    responses.
+    :param dataset: list of loaded data to analyze
+    :return: response_ad and response_story, lists of strings of all words used in user responses
+    (duplicates included)
+    """
+    # cycling through dataset to find prompt responses, sorting based on context
+    responses_ad = []
+    responses_story = []
+    question = "In three words or fewer, what is this text about?"
+    for resp in dataset:
+        # removing punctuation and making string lowercase to standardize
+        resp['response'] = resp['response'].replace('.', '').lower()
+        resp['response'] = resp['response'].replace(',', '')
+        if resp['question'] == question and resp['context'] == "This is an ad.":
+            responses_ad += resp['response'].split()
+        elif resp['question'] == question and resp['context'] == "This is actually a short story.":
+            responses_story += resp['response'].split()
+
+    return responses_ad, responses_story
+
+
+def repeated_prompt_words(dataset):
```

Comment:
This function does the same thing over two datasets basically -- each code block is exactly duplicated, with different inputs. Please refactor into a function that takes one of the two lists from your helper function as an input.